operator/internal/manifests/mutate.go

```diff
@@ -127,10 +127,9 @@ func mutateClusterRole(existing, desired *rbacv1.ClusterRole) {
 }
 
 func mutateClusterRoleBinding(existing, desired *rbacv1.ClusterRoleBinding) {
+	existing.Annotations = desired.Annotations
 	existing.Labels = desired.Labels
 	existing.Subjects = desired.Subjects
-	existing.RoleRef = desired.RoleRef
-	existing.Subjects = desired.Subjects
```

Comment:
This is probably an old hindsight, according to the official docs, roleRefs are immutable:
```
After you create a binding, you cannot change the Role or ClusterRole that it refers to. If you try to change a binding's roleRef, you get a validation error. If you do want to change the roleRef for a binding, you need to remove the binding object and create a replacement.
```
Source: https://kubernetes.io/docs/reference/access-authn-authz/rbac/#clusterrolebinding-example